### PR TITLE
git_utils.go: improve code coverage

### PR DESCRIPTION
**Pull request by skoh7645**
Bumps code coverage of `git_utils.go` from 80% to 91%
I had to capitalise some functions to export them for testing

Fixes: #717

### CODE REVIEW

File: cmd/git_utils_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+func TestStringBefore(t *testing.T) {
+	var StringBeforeTests = []struct {
+		testName       string
+		value          string
+		searchString   string
+		expectedOutput string
+	}{
+		{"Empty values", "", "", ""},
+		{"Non-empty values", "teststring", "string", "test"},
```
Comment (EnriqueL8) [inline cmd/git_utils_test.go]: one more test where the search string is not in the string value, should return empty string

File: cmd/git_utils_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+func TestGetGitInfoWithNotAGitRepo(t *testing.T) {
+	sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
+	defer cleanup()
+
+	var outBuffer bytes.Buffer
+	loggingConfig := &cmd.LoggingConfig{}
+	loggingConfig.InitLogging(&outBuffer, &outBuffer)
+	config := &cmd.RootCommandConfig{LoggingConfig: loggingConfig}
+	cwd, err := os.Getwd()
```
Comment (EnriqueL8) [inline cmd/git_utils_test.go]: Will need to investigate further on this sandbox and chang dir issue?

Comment (upLukeWeston) [reply cmd/git_utils_test.go]: Done!

File: cmd/git_utils_test.go (hunk 1 above)
Comment (upLukeWeston) [reply cmd/git_utils_test.go]: Should it return an empty string? Because when I just tested it, it returns the `value` string in full. 

For example, in this case
```
output := cmd.StringBefore("This is what I have", "But this is what I am looking for")
```
`output` = "This is what I have"
